server/src/main/java/org/opensearch/index/engine/TranslogLeafReader.java

```diff
@@ -221,28 +221,33 @@ public int maxDoc() {
 
     @Override
     public void document(int docID, StoredFieldVisitor visitor) throws IOException {
-        if (docID != 0) {
```

Comment:
Lets this implemenation be same as is and only add `storeField` implementation ?

Reply:
As `TranslogLeafReader` has it own storedFields/document implementation,  so i put the logic into `storedFields`. and implement  `StoredFields#document`  which is new lucene interface required  see the following part
https://github.com/opensearch-project/OpenSearch/blob/c18cd7fb34f7a37c56ce26608073ec75f9600023/server/src/main/java/org/opensearch/index/engine/TranslogLeafReader.java#L229-L231